test/e2e/testenv/testenv.go

```diff
@@ -0,0 +1,659 @@
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            "validator",
+							Image:           "alpine/curl",
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Command:         []string{"/bin/sh", "-c"},
+							Args:            []string{getHealthCheckScript(etcdClientServiceName, etcdClientServicePort)},
+							VolumeMounts: []corev1.VolumeMount{
+								{MountPath: "/var/etcd/ssl/ca", Name: "client-url-ca-etcd"},
+								{MountPath: "/var/etcd/ssl/server", Name: "client-url-etcd-server-tls"},
+								{MountPath: "/var/etcd/ssl/client", Name: "client-url-etcd-client-tls", ReadOnly: true},
+							},
+							ReadinessProbe: getProbe("/tmp/healthy"),
+							LivenessProbe:  getProbe("/tmp/healthy"),
+						},
+					},
+					RestartPolicy: corev1.RestartPolicyNever,
+					Volumes: []corev1.Volume{
+						getTLSVolume("client-url-ca-etcd", etcdClientTLS.TLSCASecretRef.Name),
+						getTLSVolume("client-url-etcd-server-tls", etcdClientTLS.ServerTLSSecretRef.Name),
+						getTLSVolume("client-url-etcd-client-tls", etcdClientTLS.ClientTLSSecretRef.Name),
+					},
+				},
+			},
+			BackoffLimit: ptr.To[int32](0),
+		},
+	}
+}
+
+// getHealthCheckScript generates the shell script used to check Etcd health.
+func getHealthCheckScript(etcdClientServiceName string, etcdClientServicePort int32) string {
+	return fmt.Sprintf(`failed=0; threshold=2;
+    while true; do
+        if ! curl --cacert /var/etcd/ssl/ca/ca.crt --cert /var/etcd/ssl/client/tls.crt --key /var/etcd/ssl/client/tls.key https://%s:%d/health -s -f -o /dev/null; then
+            echo "etcd is unhealthy, retrying"
+            failed=$((failed + 1))
+            if [ "$failed" -ge "$threshold" ]; then
+                echo "etcd health check failed too many times"
+                rm -f /tmp/healthy
+                exit 1
+            fi
+            sleep 2
+        else
+            echo "etcd is healthy"
+            touch /tmp/healthy
+            failed=0
+            sleep 2
+        fi
+    done`, etcdClientServiceName, etcdClientServicePort)
+}
+
+// getProbe creates a probe with specified file path for readiness and liveness checks.
+func getProbe(filePath string) *corev1.Probe {
+	return &corev1.Probe{
+		InitialDelaySeconds: 5,
+		FailureThreshold:    1,
+		PeriodSeconds:       1,
+		SuccessThreshold:    1,
+		ProbeHandler: corev1.ProbeHandler{
+			Exec: &corev1.ExecAction{
+				Command: []string{"cat", filePath},
+			},
+		},
+	}
+}
+
+// DeployEtcdLoaderJob deploys the etcd loader job.
+func (t *TestEnvironment) DeployEtcdLoaderJob(g *WithT, namespace, etcdClientServiceName string, etcdClientServicePort int32, etcdClientTLS *druidv1alpha1.TLSConfig, numKeys int64, timeout time.Duration) {
+	etcdLoaderJob, err := getEtcdLoaderJob(g, namespace, etcdClientServiceName, etcdClientServicePort, etcdClientTLS, numKeys)
+	g.Expect(err).ShouldNot(HaveOccurred())
+	g.Expect(t.cl.Create(t.ctx, etcdLoaderJob)).ShouldNot(HaveOccurred())
```

Comment:
```suggestion
	g.Expect(t.cl.Create(t.ctx, etcdLoaderJob)).To(Succeed())
```
To make the intent more clear